azext_iot/common/certops.py

```diff
@@ -32,22 +32,23 @@ def create_self_signed_certificate(subject, valid_days, cert_output_dir):
         key_file = subject + '-key.pem'
 
         open(join(cert_output_dir, cert_file), "wt").write(cert_dump)
-        open(join(cert_output_dir, key_file), "wt").write(key_dump)
+
+        if not cert_only:
+            open(join(cert_output_dir, key_file), "wt").write(key_dump)
 
     return {
         'certificate': cert_dump,
         'privateKey': key_dump,
-        'thumbprint': thumbprint
+        'thumbprint': thumbprint,
     }
 
-
 def open_certificate(certificate_path):
     certificate = ""
     if certificate_path.endswith('.pem') or certificate_path.endswith('.cer'):
         with open(certificate_path, "rb") as cert_file:
             certificate = cert_file.read()
             try:
-                certificate = certificate.decode("utf-8")
+                certificate = certificate.decode("utf-8").rstrip("\r\n")
```

Comment:
For understanding, what is the benefit of this?

Reply:
This is a service bug, that they cannot handle certificate containing a \r\n as the ending. They are fixing it, but would take a while to deploy, so I am adding this as a work around. I will double check if they have already fixed it before we release the CLI